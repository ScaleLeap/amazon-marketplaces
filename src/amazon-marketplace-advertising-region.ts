export interface AmazonMarketplaceAdvertisingRegion {
  /**
   * Region code.
   */
  code: string

  /**
   * Region name.
   */
  name: string

  /**
   * Authorization URI.
   */
  authorizationUri: string

  /**
   * Access Token URI
   */
  accessTokenUri: string

  /**
   * Access endpoint URI.
   */
  endpoint: string
}

export class AmazonMarketplaceAdvertisingRegion implements AmazonMarketplaceAdvertisingRegion {
  constructor(data: AmazonMarketplaceAdvertisingRegion) {
    Object.assign(this, data)
    Object.freeze(this)
  }
}
