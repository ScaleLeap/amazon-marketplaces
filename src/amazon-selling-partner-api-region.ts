export interface AmazonSellingPartnerApiRegion {
  /**
   * Region code.
   */
  code: string

  /**
   * Region name.
   */
  name: string

  /**
   * Access endpoint URI.
   */
  endpoint: string

  /**
   * AWS region.
   */
  awsRegion: string
}

export class AmazonSellingPartnerApiRegion implements AmazonSellingPartnerApiRegion {
  constructor(data: AmazonSellingPartnerApiRegion) {
    Object.assign(this, data)
    Object.freeze(this)
  }
}
