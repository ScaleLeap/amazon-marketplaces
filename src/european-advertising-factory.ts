import {
  AmazonMarketplaceAdvertisingCountryCode,
  AmazonMarketplaceAdvertisingTimeZone,
  AmazonMarketplaceAdvertising,
} from './amazon-marketplace'
import { marketplaceAdvertisingRegions } from './marketplace-advertising-regions'

export function europeanAdvertisingFactory(
  countryCode: AmazonMarketplaceAdvertisingCountryCode,
): AmazonMarketplaceAdvertising {
  return {
    countryCode,
    region: marketplaceAdvertisingRegions.EU,
    bids: {
      sponsoredBrands: {
        min: 10,
        max: 3900,
      },
      sponsoredProducts: {
        min: 2,
        max: 100000,
      },
    },
    timeZone: AmazonMarketplaceAdvertisingTimeZone.EUROPE_PARIS,
  }
}
