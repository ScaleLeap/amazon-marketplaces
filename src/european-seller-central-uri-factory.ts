export function europeanSellerCentralUriFactory(): string {
  return 'https://sellercentral-europe.amazon.com'
}
