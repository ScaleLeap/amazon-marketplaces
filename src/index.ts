export * from './amazon-marketplace'
export * from './marketplaces'
export * from './utils'
export * from './amazon-selling-partner-api-region'
export * from './selling-partner-api-regions'
