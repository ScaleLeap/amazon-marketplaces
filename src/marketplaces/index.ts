import { AE } from './ae'
import { AU } from './au'
import { BE } from './be'
import { BR } from './br'
import { CA } from './ca'
import { CN } from './cn'
import { DE } from './de'
import { EG } from './eg'
import { ES } from './es'
import { FR } from './fr'
import { GB } from './gb'
import { IN } from './in'
import { IT } from './it'
import { JP } from './jp'
import { MX } from './mx'
import { NL } from './nl'
import { PL } from './pl'
import { SA } from './sa'
import { SE } from './se'
import { SG } from './sg'
import { TR } from './tr'
import { US } from './us'

export const amazonMarketplaces = {
  /**
   * United Arab Emirates
   */
  AE,

  /**
   * Australia
   */
  AU,

  /**
   * Belgium
   */
  BE,

  /**
   * Brazil
   */
  BR,

  /**
   * Canada
   */
  CA,

  /**
   * China
   */
  CN,

  /**
   * Germany
   */
  DE,

  /**
   * Egypt
   */
  EG,

  /**
   * Spain
   */
  ES,

  /**
   * France
   */
  FR,

  /**
   * United Kingdom
   */
  GB,

  /**
   * India
   */
  IN,

  /**
   * Italy
   */
  IT,

  /**
   * Japan
   */
  JP,

  /**
   * Mexico
   */
  MX,

  /**
   * Netherlands
   */
  NL,

  /**
   * Poland
   */
  PL,

  /**
   * Singapore
   */
  SA,

  /**
   * Sweden
   */
  SE,

  /**
   * Singapore
   */
  SG,

  /**
   * Turkey
   */
  TR,

  /**
   * United States
   */
  US,
} as const

/**
 * An array of all of the `AmazonMarketplace` objects.
 */
export const amazonMarketplacesList = Object.values(amazonMarketplaces)
