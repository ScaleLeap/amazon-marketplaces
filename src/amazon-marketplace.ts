import { AmazonMarketplaceAdvertisingRegion } from './amazon-marketplace-advertising-region'

/**
 * Amazon Marketplace country code enum.
 */
export enum AmazonMarketplaceCountryCode {
  AE = 'AE',
  AU = 'AU',
  BR = 'BR',
  CA = 'CA',
  CN = 'CN',
  DE = 'DE',
  ES = 'ES',
  FR = 'FR',
  GB = 'GB',
  IN = 'IN',
  IT = 'IT',
  JP = 'JP',
  MX = 'MX',
  NL = 'NL',
  SG = 'SG',
  TR = 'TR',
  US = 'US',
}

/**
 * Currency enum.
 */
export enum AmazonMarketplaceAdvertisingCurrency {
  AED = 'AED',
  AUD = 'AUD',
  BRL = 'BRL',
  CAD = 'CAD',
  CNY = 'CNY',
  EUR = 'EUR',
  GBP = 'GBP',
  INR = 'INR',
  JPY = 'JPY',
  MXN = 'MXN',
  SGD = 'SGD',
  TRY = 'TRY',
  USD = 'USD',
}

/**
 * Amazon Marketplace Advertising country code enum.
 *
 * The country code identifying the publisher(s) on which ads will run.
 */
export enum AmazonMarketplaceAdvertisingCountryCode {
  AE = 'AE',
  AU = 'AU',
  CA = 'CA',
  DE = 'DE',
  ES = 'ES',
  FR = 'FR',
  IT = 'IT',
  JP = 'JP',
  UK = 'UK', // Not a real country code, but such is life. See https://github.com/ScaleLeap/amazon-marketplaces/issues/122
  US = 'US',
}

/**
 * Time Zone enum.
 */
export enum AmazonMarketplaceAdvertisingTimeZone {
  AMERICA_LOS_ANGELES = 'America/Los_Angeles',
  ASIA_DUBAI = 'Asia/Dubai',
  ASIA_TOKYO = 'Asia/Tokyo',
  AUSTRALIA_SYDNEY = 'Australia/Sydney',
  EUROPE_LONDON = 'Europe/London',
  EUROPE_PARIS = 'Europe/Paris',
}

export interface AmazonMarketplaceAdvertising {
  /**
   * Amazon Advertising account time zone.
   *
   * @example
   *
   * "America/Los_Angeles"
   */
  readonly timeZone: AmazonMarketplaceAdvertisingTimeZone

  /**
   * The country code identifying the publisher(s) on which ads will run.
   *
   * [ISO 3166-1](https://en.wikipedia.org/wiki/ISO_3166-1) with the exception of UK, which uses
   * `UK` instead of `GB`.
   *
   * @example
   *
   * AmazonMarketplaceAdvertisingCountryCode.CA
   */
  readonly countryCode: AmazonMarketplaceAdvertisingCountryCode

  /**
   * Amazon Advertising API Regions
   */
  readonly region: AmazonMarketplaceAdvertisingRegion

  /**
   * Minimum and maximum bids for each advertising type.
   */
  readonly bids: {
    /**
     * Sponsored Products (SP)
     */
    readonly sponsoredProducts: AmazonAdvertisingBidLimits

    /**
     * Sponsored Brands (SB)
     */
    readonly sponsoredBrands: AmazonAdvertisingBidLimits
  }
}

export interface AmazonAdvertisingBidLimits {
  /**
   * Minimum bid in account currency in **smallest units** (e.g. cents).
   *
   * @example
   *
   * 2
   */
  readonly min: number

  /**
   * Maximum bid in account currency in **smallest units** (e.g. cents).
   *
   * @example
   *
   * 100000
   */
  readonly max: number
}

export interface AmazonMarketplace {
  /**
   * Amazon Marketplace ID.
   *
   * @example
   *
   * 'A2EUQ1WTGCTBG2'
   */
  readonly id: string

  /**
   * Amazon Marketplace Name.
   *
   * @example
   *
   * 'Canada'
   */
  readonly name: string

  /**
   * Country code for the given marketplace.
   *
   * [ISO 3166-1](https://en.wikipedia.org/wiki/ISO_3166-1)
   *
   * @example
   *
   * AmazonMarketplaceCountryCode.CA
   */
  readonly countryCode: AmazonMarketplaceCountryCode

  /**
   * Currency code for the given marketplace.
   *
   * [ISO 4217](https://en.wikipedia.org/wiki/ISO_4217#X_currencies)
   *
   * @example
   *
   * AmazonMarketplaceAdvertisingCurrency.CAD
   */
  readonly currency: AmazonMarketplaceAdvertisingCurrency

  /**
   * Amazon Marketplace URI.
   *
   * @example
   *
   * 'https://www.amazon.ca/'
   */
  readonly uri: string

  /**
   * [Amazon Marketplace Web Service](https://developer.amazonservices.com/gp/mws/docs.html) (Amazon MWS) Endpoint URI.
   *
   * @example
   *
   * 'https://mws.amazonservices.ca'
   */
  readonly webServiceUri: string

  /**
   * [Amazon Advertising](https://advertising.amazon.com/API/docs/v2/guides/supported_features).
   */
  readonly advertising?: AmazonMarketplaceAdvertising
}

export class AmazonMarketplace implements AmazonMarketplace {
  public constructor(amazonMarketplace: AmazonMarketplace) {
    Object.assign(this, amazonMarketplace)
    Object.freeze(this)
  }
}
