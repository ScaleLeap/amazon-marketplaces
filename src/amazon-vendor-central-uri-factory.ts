import { AmazonMarketplaceCountryCode } from './amazon-marketplace'

export function amazonVendorCentralUriFactory(countryCode: AmazonMarketplaceCountryCode): string {
  return `https://vendorcentral.amazon.${countryCode.toLowerCase()}`
}
