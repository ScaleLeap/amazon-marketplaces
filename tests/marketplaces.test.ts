import { amazonMarketplaces } from '../src/marketplaces'
import { readdirSync } from 'fs'
import { join } from 'path'

const countryCodes = readdirSync(join(__dirname, '../src/marketplaces'))
  .map((f) => f.split('.')[0])
  .filter((f) => f.match(/^[A-Z]{2}$/))

describe('marketplace', () => {
  it('has a known number of marketplaces', () => {
    expect.assertions(1)

    expect(countryCodes.length).toBe(17)
  })

  describe.each(countryCodes)('%s', (countryCode) => {
    const code = (countryCode as unknown) as keyof typeof amazonMarketplaces

    it('should match the exports', () => {
      expect.assertions(1)

      expect(amazonMarketplaces[code].countryCode).toBe(code)
    })

    it('should match snapshot', () => {
      expect.assertions(1)

      expect(amazonMarketplaces[code]).toMatchSnapshot()
    })
  })

  it('object is read only', () => {
    expect(() => Object.assign(amazonMarketplaces.CA, { id: 'a' })).toThrowError(/Cannot assign to read only property/)
  })
})
