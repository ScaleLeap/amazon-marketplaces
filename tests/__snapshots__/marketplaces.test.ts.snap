// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`marketplace AE should match snapshot 1`] = `
AmazonMarketplace {
  "advertising": Object {
    "bids": Object {
      "sponsoredBrands": Object {
        "max": 18400,
        "min": 24,
      },
      "sponsoredProducts": Object {
        "max": 18400,
        "min": 24,
      },
    },
    "countryCode": "AE",
    "region": AmazonMarketplaceAdvertisingRegion {
      "accessTokenUri": "https://api.amazon.co.uk/auth/o2/token",
      "authorizationUri": "https://eu.account.amazon.com/ap/oa",
      "code": "EU",
      "endpoint": "https://advertising-api-eu.amazon.com",
      "name": "Europe",
    },
    "timeZone": "Asia/Dubai",
  },
  "countryCode": "AE",
  "currency": "AED",
  "id": "A2VIGQ35RCS4UG",
  "name": "United Arab Emirates",
  "uri": "https://www.amazon.ae",
  "webServiceUri": "https://mws.amazonservices.ae",
}
`;

exports[`marketplace AU should match snapshot 1`] = `
AmazonMarketplace {
  "advertising": Object {
    "bids": Object {
      "sponsoredBrands": Object {
        "max": 7000,
        "min": 10,
      },
      "sponsoredProducts": Object {
        "max": 141000,
        "min": 10,
      },
    },
    "countryCode": "AU",
    "region": AmazonMarketplaceAdvertisingRegion {
      "accessTokenUri": "https://api.amazon.co.jp/auth/o2/token",
      "authorizationUri": "https://apac.account.amazon.com/ap/oa",
      "code": "FE",
      "endpoint": "https://advertising-api-fe.amazon.com",
      "name": "Far East",
    },
    "timeZone": "Australia/Sydney",
  },
  "countryCode": "AU",
  "currency": "AUD",
  "id": "A39IBJ37TRP1C6",
  "name": "Australia",
  "uri": "https://www.amazon.com.au",
  "webServiceUri": "https://mws.amazonservices.com.au",
}
`;

exports[`marketplace BR should match snapshot 1`] = `
AmazonMarketplace {
  "countryCode": "BR",
  "currency": "BRL",
  "id": "A2Q3Y263D00KWC",
  "name": "Brazil",
  "uri": "https://www.amazon.com.br",
  "webServiceUri": "https://mws.amazonservices.com",
}
`;

exports[`marketplace CA should match snapshot 1`] = `
AmazonMarketplace {
  "advertising": Object {
    "bids": Object {
      "sponsoredBrands": Object {
        "max": 4900,
        "min": 10,
      },
      "sponsoredProducts": Object {
        "max": 100000,
        "min": 2,
      },
    },
    "countryCode": "CA",
    "region": AmazonMarketplaceAdvertisingRegion {
      "accessTokenUri": "https://api.amazon.com/auth/o2/token",
      "authorizationUri": "https://www.amazon.com/ap/oa",
      "code": "NA",
      "endpoint": "https://advertising-api.amazon.com",
      "name": "North America",
    },
    "timeZone": "America/Los_Angeles",
  },
  "countryCode": "CA",
  "currency": "CAD",
  "id": "A2EUQ1WTGCTBG2",
  "name": "Canada",
  "uri": "https://www.amazon.ca",
  "webServiceUri": "https://mws.amazonservices.ca",
}
`;

exports[`marketplace CN should match snapshot 1`] = `
AmazonMarketplace {
  "countryCode": "CN",
  "currency": "CNY",
  "id": "AAHKV2X7AFYLW",
  "name": "China",
  "uri": "https://www.amazon.cn",
  "webServiceUri": "https://mws.amazonservices.com.cn",
}
`;

exports[`marketplace DE should match snapshot 1`] = `
AmazonMarketplace {
  "advertising": Object {
    "bids": Object {
      "sponsoredBrands": Object {
        "max": 3900,
        "min": 10,
      },
      "sponsoredProducts": Object {
        "max": 100000,
        "min": 2,
      },
    },
    "countryCode": "DE",
    "region": AmazonMarketplaceAdvertisingRegion {
      "accessTokenUri": "https://api.amazon.co.uk/auth/o2/token",
      "authorizationUri": "https://eu.account.amazon.com/ap/oa",
      "code": "EU",
      "endpoint": "https://advertising-api-eu.amazon.com",
      "name": "Europe",
    },
    "timeZone": "Europe/Paris",
  },
  "countryCode": "DE",
  "currency": "EUR",
  "id": "A1PA6795UKMFR9",
  "name": "Germany",
  "uri": "https://www.amazon.de",
  "webServiceUri": "https://mws-eu.amazonservices.com",
}
`;

exports[`marketplace ES should match snapshot 1`] = `
AmazonMarketplace {
  "advertising": Object {
    "bids": Object {
      "sponsoredBrands": Object {
        "max": 3900,
        "min": 10,
      },
      "sponsoredProducts": Object {
        "max": 100000,
        "min": 2,
      },
    },
    "countryCode": "ES",
    "region": AmazonMarketplaceAdvertisingRegion {
      "accessTokenUri": "https://api.amazon.co.uk/auth/o2/token",
      "authorizationUri": "https://eu.account.amazon.com/ap/oa",
      "code": "EU",
      "endpoint": "https://advertising-api-eu.amazon.com",
      "name": "Europe",
    },
    "timeZone": "Europe/Paris",
  },
  "countryCode": "ES",
  "currency": "EUR",
  "id": "A1RKKUPIHCS9HS",
  "name": "Spain",
  "uri": "https://www.amazon.es",
  "webServiceUri": "https://mws-eu.amazonservices.com",
}
`;

exports[`marketplace FR should match snapshot 1`] = `
AmazonMarketplace {
  "advertising": Object {
    "bids": Object {
      "sponsoredBrands": Object {
        "max": 3900,
        "min": 10,
      },
      "sponsoredProducts": Object {
        "max": 100000,
        "min": 2,
      },
    },
    "countryCode": "FR",
    "region": AmazonMarketplaceAdvertisingRegion {
      "accessTokenUri": "https://api.amazon.co.uk/auth/o2/token",
      "authorizationUri": "https://eu.account.amazon.com/ap/oa",
      "code": "EU",
      "endpoint": "https://advertising-api-eu.amazon.com",
      "name": "Europe",
    },
    "timeZone": "Europe/Paris",
  },
  "countryCode": "FR",
  "currency": "EUR",
  "id": "A13V1IB3VIYZZH",
  "name": "France",
  "uri": "https://www.amazon.fr",
  "webServiceUri": "https://mws-eu.amazonservices.com",
}
`;

exports[`marketplace GB should match snapshot 1`] = `
AmazonMarketplace {
  "advertising": Object {
    "bids": Object {
      "sponsoredBrands": Object {
        "max": 3100,
        "min": 10,
      },
      "sponsoredProducts": Object {
        "max": 100000,
        "min": 2,
      },
    },
    "countryCode": "UK",
    "region": AmazonMarketplaceAdvertisingRegion {
      "accessTokenUri": "https://api.amazon.co.uk/auth/o2/token",
      "authorizationUri": "https://eu.account.amazon.com/ap/oa",
      "code": "EU",
      "endpoint": "https://advertising-api-eu.amazon.com",
      "name": "Europe",
    },
    "timeZone": "Europe/London",
  },
  "countryCode": "GB",
  "currency": "GBP",
  "id": "A1F83G8C2ARO7P",
  "name": "United Kingdom",
  "uri": "https://www.amazon.co.uk",
  "webServiceUri": "https://mws-eu.amazonservices.com",
}
`;

exports[`marketplace IN should match snapshot 1`] = `
AmazonMarketplace {
  "countryCode": "IN",
  "currency": "INR",
  "id": "A21TJRUUN4KGV",
  "name": "India",
  "uri": "https://www.amazon.in",
  "webServiceUri": "https://mws.amazonservices.in",
}
`;

exports[`marketplace IT should match snapshot 1`] = `
AmazonMarketplace {
  "advertising": Object {
    "bids": Object {
      "sponsoredBrands": Object {
        "max": 3900,
        "min": 10,
      },
      "sponsoredProducts": Object {
        "max": 100000,
        "min": 2,
      },
    },
    "countryCode": "DE",
    "region": AmazonMarketplaceAdvertisingRegion {
      "accessTokenUri": "https://api.amazon.co.uk/auth/o2/token",
      "authorizationUri": "https://eu.account.amazon.com/ap/oa",
      "code": "EU",
      "endpoint": "https://advertising-api-eu.amazon.com",
      "name": "Europe",
    },
    "timeZone": "Europe/Paris",
  },
  "countryCode": "IT",
  "currency": "EUR",
  "id": "APJ6JRA9NG5V4",
  "name": "Italy",
  "uri": "https://www.amazon.it",
  "webServiceUri": "https://mws-eu.amazonservices.com",
}
`;

exports[`marketplace JP should match snapshot 1`] = `
AmazonMarketplace {
  "advertising": Object {
    "bids": Object {
      "sponsoredBrands": Object {
        "max": 776000,
        "min": 1000,
      },
      "sponsoredProducts": Object {
        "max": 10000000,
        "min": 200,
      },
    },
    "countryCode": "JP",
    "region": AmazonMarketplaceAdvertisingRegion {
      "accessTokenUri": "https://api.amazon.co.jp/auth/o2/token",
      "authorizationUri": "https://apac.account.amazon.com/ap/oa",
      "code": "FE",
      "endpoint": "https://advertising-api-fe.amazon.com",
      "name": "Far East",
    },
    "timeZone": "Asia/Tokyo",
  },
  "countryCode": "JP",
  "currency": "JPY",
  "id": "A1VC38T7YXB528",
  "name": "Japan",
  "uri": "https://www.amazon.co.jp",
  "webServiceUri": "https://mws.amazonservices.jp",
}
`;

exports[`marketplace MX should match snapshot 1`] = `
AmazonMarketplace {
  "countryCode": "MX",
  "currency": "MXN",
  "id": "A1AM78C64UM0Y8",
  "name": "Mexico",
  "uri": "https://www.amazon.com.mx",
  "webServiceUri": "https://mws.amazonservices.com.mx",
}
`;

exports[`marketplace NL should match snapshot 1`] = `
AmazonMarketplace {
  "countryCode": "NL",
  "currency": "EUR",
  "id": "A1805IZSGTT6HS",
  "name": "Netherlands",
  "uri": "https://www.amazon.nl",
  "webServiceUri": "https://mws-eu.amazonservices.com",
}
`;

exports[`marketplace SG should match snapshot 1`] = `
AmazonMarketplace {
  "countryCode": "SG",
  "currency": "SGD",
  "id": "A19VAU5U5O7RUS",
  "name": "Singapore",
  "uri": "https://www.amazon.sg",
  "webServiceUri": "https://mws-fe.amazonservices.com",
}
`;

exports[`marketplace TR should match snapshot 1`] = `
AmazonMarketplace {
  "countryCode": "TR",
  "currency": "TRY",
  "id": "A33AVAJ2PDY3EV",
  "name": "Turkey",
  "uri": "https://www.amazon.com.tr",
  "webServiceUri": "https://mws-eu.amazonservices.com",
}
`;

exports[`marketplace US should match snapshot 1`] = `
AmazonMarketplace {
  "advertising": Object {
    "bids": Object {
      "sponsoredBrands": Object {
        "max": 4900,
        "min": 10,
      },
      "sponsoredProducts": Object {
        "max": 100000,
        "min": 2,
      },
    },
    "countryCode": "US",
    "region": AmazonMarketplaceAdvertisingRegion {
      "accessTokenUri": "https://api.amazon.com/auth/o2/token",
      "authorizationUri": "https://www.amazon.com/ap/oa",
      "code": "NA",
      "endpoint": "https://advertising-api.amazon.com",
      "name": "North America",
    },
    "timeZone": "America/Los_Angeles",
  },
  "countryCode": "US",
  "currency": "USD",
  "id": "ATVPDKIKX0DER",
  "name": "United States",
  "uri": "https://www.amazon.com",
  "webServiceUri": "https://mws.amazonservices.com",
}
`;
